### PR TITLE
hedgedoc: fix editor crashing 

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by winterqt** [comment]
So did upstream do away with esbuild? Seems like it since you didn't have to add our esbuild.

**Comment by SuperSandro2000** [comment]
No, we are just using the one in node_modules. It does not make sense to vendor it anyway because it must match the lock file.

**Comment by winterqt** [comment]
> No, we are just using the one in node_modules. It does not make sense to vendor it anyway because it must match the lock file.

Wonder why we didn't use the one in node_modules all this time then.

File: pkgs/servers/web-apps/hedgedoc/default.nix [hunk 1]
```diff
@@ -76,10 +61,8 @@ mkYarnPackage rec {
...
     popd
 
-    pushd node_modules/esbuild
-    rm bin/esbuild
-    ln -s ${lib.getBin esbuild-hedgedoc}/bin/esbuild bin/
-    popd
+    # TODO: debug why this file is not included in the upstream webpack bundle.
+    rm node_modules/js-yaml/dist/js-yaml.mjs
 
     npm run build
```
Comment (drupol) [inline pkgs/servers/web-apps/hedgedoc/default.nix:67]: How about using yarn in here ?

**Comment by winterqt** [comment]
I changed the title of this PR to remove the leading space. Amazingly, GitHub doesn't trim the PR title upon creation...? @SuperSandro2000 Did you use the website to make this PR? Normally your browser should strip it, so I'm guessing this was made through the API?

**Comment by drupol** [comment]
Thanks for taking care of that @SuperSandro2000 !

I tried your branch locally and it works. I also tried to replace `npm run build` with `yarn build` and works as well.

Don't you think we should take the opportunity to update the build script and use yarn for building?

Check the [official installation procedure](https://docs.hedgedoc.org/setup/manual-setup/), I think we should follow those steps.

**Review by drupol — APPROVED** [review]

File: pkgs/servers/web-apps/hedgedoc/default.nix (hunk 1 above)
Comment (drupol) [inline pkgs/servers/web-apps/hedgedoc/default.nix:67]:
```suggestion
    yarn build
```

**Comment by drupol** [comment]
I updated my hedgedoc server with the changes from here, and it works. However, when going to the features page, there is an message on the top now.



I guess this is related to the file we removed.

**Comment by SuperSandro2000** [comment]
> Did you use the website to make this PR?

Probably something off in Refinded GitHub.



> I tried your branch locally and it works. I also tried to replace `npm run build` with `yarn build` and works as well.

I tried that, too and it didn't out of the box. yarn is a bit more tricky to handle and run offline so I would ignore it for now. As mentioned earlier this just runs the build script and the program that is used shouldn't matter.

File: pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix [hunk 2]
```diff
@@ -69,7 +69,7 @@ in rec {
     yarnLock,
     yarnNix ? mkYarnNix { inherit yarnLock; },
     offlineCache ? importOfflineCache yarnNix,
-    yarnFlags ? defaultYarnFlags,
+    yarnFlags ? [ ],
```
Comment (SuperSandro2000) [inline pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:72]: This is not much used in nixpkgs and I am pretty confident that this is a none breaking change. I am still going to remove the extra flags accumulated in other packages after this is merged.

**Review by drupol — APPROVED** [review]
I couldn't agree more with this PR.

We should also fix broken HedgeDoc 1.9.4 so I think we should merge it asap.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 182643` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>hedgedoc</li>
<li>uivonim</li>
<li>vieb</li>
<li>yarn2nix</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
hedgedoc:
<details>

When evaluating attribute ‘hedgedoc’:
warning: EvalError
Cannot evaluate attribute ‘hedgedoc’ in ‘/nix/store/7n5k8glkfw19zn14g0apnrl507sm155l-nixpkgs’.
</details>
uivonim:
<details>

When evaluating attribute ‘uivonim’:
warning: EvalError
Cannot evaluate attribute ‘uivonim’ in ‘/nix/store/7n5k8glkfw19zn14g0apnrl507sm155l-nixpkgs’.
</details>
vieb:
<details>

When evaluating attribute ‘vieb’:
warning: EvalError
Cannot evaluate attribute ‘vieb’ in ‘/nix/store/7n5k8glkfw19zn14g0apnrl507sm155l-nixpkgs’.
</details>
yarn2nix:
<details>

When evaluating attribute ‘yarn2nix’:
warning: EvalError
Cannot evaluate attribute ‘yarn2nix’ in ‘/nix/store/7n5k8glkfw19zn14g0apnrl507sm155l-nixpkgs’.
</details>
</details>

**Comment by drupol** [comment]
Excellent, thanks !!!

File: pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix (hunk 2 above)
Comment (thatsmydoing) [reply pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:72]: We use this in our internal project to remove the `--ignore-scripts` flag when we need to build native extensions for a module. As it is now, it doesn't seem possible to unset any of the default flags so our use case is broken (what's more it's been backported to nixos-22.05).

Comment (SuperSandro2000) [reply pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:72]: Not sure how we build those in nixpkgs. How about adding a boolean variable wether to run scripts?

Comment (SuperSandro2000) [reply pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:72]: #198950

**Pull request merged**